src/routes/tasks.rs

```diff
@@ -51,26 +49,27 @@ impl FromConn for Task {
     }
 }
 
-type CreateArgs = (Account, Json<NewTask>, AggregatorClient, Db);
+type CreateArgs = (Account, Json<NewTask>, State<Client>, Db);
 pub async fn create(
-    conn: &mut Conn,
-    (account, Json(task), api_client, db): CreateArgs,
+    _: &mut Conn,
+    (account, task, State(client), db): CreateArgs,
 ) -> Result<impl Handler, Error> {
-    task.validate()?;
-    let config = conn.state().ok_or(Error::NotFound)?;
-    let task_create = TaskCreate::build(task.clone(), config)?;
-    let api_response = api_client.create_task(task_create).await?;
-    let task = build_task(task, api_response, &account).insert(&db).await?;
-    Ok((Status::Created, Json(task)))
+    task.validate(account, &db)
+        .await?
+        .provision(client)
+        .await?
+        .insert(&db)
+        .await
+        .map_err(Into::into)
+        .map(|task| (Status::Created, Json(task)))
```

Comment:
AFAICT the caller gets back a JSON encoded `entity::task::Model`, which does not include the collector auth token that will be generated by the leader.

Reply:
Correct. We do not do anything with the collector auth token as of this PR